File: cpp/difference-of-squares/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares
{
    int square_of_sum(int n)
    {
        int sum = 0;

        for (int i = 1; i <= n; i++)
        {
            sum += i;
        }

        return sum * sum;
    }

    int sum_of_squares(int n)
    {
        int sum = 0;

        for (int i = 1; i <= n; i++)
        {
            sum += (i * i);
        }

        return sum;
    }

    int difference(int n)
    {
        return abs(sum_of_squares(n) - square_of_sum(n));
    }
} // namespace difference_of_squares

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize(num)
    @num = num
  end

  def square_of_sum
    sum = 0

    @num.times do |i|
      sum += i + 1
    end

    sum ** 2
  end

  def sum_of_squares
    sum = 0

    @num.times do |i|
      sum += (i + 1) ** 2
    end

    sum
  end

  def difference
    (square_of_sum() - sum_of_squares()).abs
  end
end
